.github/workflows/release.yml

```
@@ -46,10 +51,37 @@ jobs:
           echo "::set-output name=name::${GITHUB_REF#refs/*/}"
           echo "::set-output name=version::${GITHUB_REF##*/}"
         id: extract_branch
+      - name: Check npm version preid
+        run: |
+          HAS_PREID=$(node -p "'${{ steps.extract_branch.outputs.version }}'.startsWith('prerelease-') ? 1 : 0")
+          echo "::set-output name=has_preid::${HAS_PREID}"
+          if [ $HAS_PREID -eq 1 ]; then
+            PREID_NAME=$(node -p "const v = '${{ steps.extract_branch.outputs.version }}'.match(/^prerelease-(.+)$/); v ? v[1] : ''")
+            # validate preid
+            IS_ALLOWED=$(node -p "'${ALLOW_PREID}'.split('|').includes('${PREID_NAME}') ? 1 : 0")
+            if [ $IS_ALLOWED -eq 0 ]; then
```

[nits] ALLOW_PREIDに常に正規表現が入るのなら

```suggestion
            if [[ "$IS_ALLOWED" =~ $ALLOW_PREID ]]; then
```

とBashに判定させるのも良いかなと思いました（`$ALLOW_PREID`は`"`で囲むとダメです）。